README.md

```diff
@@ -9,6 +9,19 @@ This application is built as an example for [Oso Cloud](https://osohq.com/docs/)
 If you are looking for an example of using the Oso library, check out
 [osohq/gitclub](https://github.com/osohq/gitclub).
 
+## Dependencies
+Use a package manager such as `apt-get` (Linux distributions) or `Homebrew` (MacOS) to install the required and
```

Comment:
I would leave out "such as `apt-get` (Linux distibutions)", since `apt-get` is just for dpkg-based distributions (Debian, Ubuntu) and will not work on other distributions (Fedora, RHEL).

Maybe: "Use a package manager to install the required and recommended software tools and packages below." for this first update?

Reply:
> > I initially thought the package.json file was being tracked, hence the npm install command.
> > What do you all think about adding one to the repo? That will solve other issues users will face if they try to run the services without all the dependencies.
> 
> Yeah - I think we should _definitely_ have a `package.json` in this repository, assuming users need to run `npm install`. But, I think we do have `package.json` under the `frontend` directory:
> 
> * https://github.com/osohq/gitcloud/blob/main/frontend/package.json
> 
> Not sure about whether this should be top-level, I'm not as familiar with JavaScript development.

Yeah we can ask. I missed the package.json in the front end. It makes sense that it's there. Have you tried running it? Was it successful for you? I guess we can look at some of these other issues in another PR.

Reply:
I'm testing it out now.

So, first thing I see is that there's a couple scripts defined in `frontend/package.json` (`jq '.scripts' < package.json`):
```json
{
  "dev": "next dev -p 8000",
  "build": "next build",
  "start": "next start",
  "lint": "next lint"
}
```

So it looks like all of these commands are using the [`next.js` CLI](https://nextjs.org/docs/api-reference/cli) :
- `dev` will start a next.js development server:
> next dev starts the application in development mode with hot-code reloading, error reporting, and more
- `build` will create a production build that can be deployed:
> next build creates an optimized production build of your application. The output displays information about each route.
- `start` will run the application in production mode:
> next start starts the application in production mode. The application should be compiled with [next build](https://nextjs.org/docs/api-reference/cli#build) first.
- `lint` will run lints:
> next lint runs ESLint for all files in the pages, components, and lib directories. It also provides a guided setup to install any required dependencies if ESLint is not already configured in your application.

---

I can see that I'll need next.js to run this application, so first question I ask: "is it included in the dependencies, such that it would be installed with `npm install`?"

And yes, it is, version `12.3.1`:
```
❯ jq '.dependencies | .next' -c < package.json
"12.3.1"
```

So I did try running `npm install`, but it failed:
```
❯ npm install                       
npm notice                  
npm notice New major version of npm available! 8.19.1 -> 9.3.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.3.1
npm notice Run npm install -g npm@9.3.1 to update!
npm notice                    
npm ERR! code ERESOLVE 
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!                 
npm ERR! While resolving: frontend-nextjs@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.0" from use-persisted-state@0.3.3
npm ERR! node_modules/use-persisted-state
npm ERR!   use-persisted-state@"^0.3.3" from the root project
npm ERR!                         
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!            
npm ERR! See /home/djanatyn/.npm/eresolve-report.txt for a full report.
                                               
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/djanatyn/.npm/_logs/2023-01-18T15_27_47_922Z-debug-0.log
```

So, we should probably specify **the npm version that is required**, assuming this installs with a newer version of `npm`.

Reply:
Talked with @samscott89 , and it looks like we only need `yarn` here. Running `yarn` will pull down dependencies (according to `yarn.lock`), which includes `next.js`. We don't need to run `npm install` at any point, and we don't need to tell users to install `npm`. They just need `yarn` for frontend.